internal/exec/exec.go

```diff
@@ -40,12 +38,55 @@ func makePanicError(value interface{}) *errors.QueryError {
 	return errors.Errorf("graphql: panic occurred: %v", value)
 }
 
+func (r *Request) process(ctx context.Context, nodes []*execNode, s *resolvable.Schema) {
+	queue := nodes
+	for len(queue) > 0 {
+		batch := queue
+		queue = nil
+
+		// run the whole batch (everything from the current level) concurrently.
+		result := make(chan *execNode, len(batch))
+		for _, n := range batch {
+			go func(n *execNode) {
+				if n.field != nil {
+					r.resolveNode(ctx, n, true)
+					expandNode(n, s)
+				}
+				result <- n
+			}(n)
+		}
+
+		// wait for the batch to complete and refill the queue.
+		for range batch {
+			n := <-result
+			queue = append(queue, n.children...)
+		}
+	}
+}
+
 func (r *Request) Execute(ctx context.Context, s *resolvable.Schema, op *query.Operation) ([]byte, []*errors.QueryError) {
-	var out bytes.Buffer
+	out := &bytes.Buffer{}
```

Comment:
Please, use the default values. This needs to be changed back to `var out bytes.Buffer` the zero value is useful as stated by Rob Pike in his "[Make the zero value useful](https://go-proverbs.github.io/)." proverb.